website/server/models/group.js

```diff
@@ -750,15 +750,18 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all') {
     $inc: {memberCount: -1},
   };
 
-  let challenges = await Challenge.find({
-    _id: {$in: user.challenges},
-    group: group._id,
-  });
+  // only remove user from challenges if it's set to remove-all
+  if (keep === 'remove-all') {
```

Comment:
I think the API docs for the group leave route need to be updated https://github.com/HabitRPG/habitrpg/blob/b54441a637b91ddd80e370cd78885514c8056b3e/website/server/controllers/api-v3/groups.js#L362-L372

Since `keep` no longer defaults to anything, right? We should also clarify better what the keep queryParam means when passing it in.